### PR TITLE
Use commit's sha when known as the chekout folder name

### LINKED ISSUE #699: Build directory is being created in wrong places
After a manual trigger being done on one of my example repos I end up with the following `~/.fastlane/ci` directory tree:
```
.
├── 5bf2a0f7-cfdb-43f0-aa0d-065e76e53bc8
│   ├── Gemfile
│   ├── Gemfile.lock
│   ├── README.md
│   ├── builds
│   ├── fastlane
│   ├── fastlane-ci-demoapp
│   ├── fastlane-ci-demoapp.xcodeproj
│   └── manual_build_8fa0ecfb-cc15-4693-9839-b6c1632ae095
├── artifacts
│   └── 5bf2a0f7-cfdb-43f0-aa0d-065e76e53bc8
├── fastlane-ci-config
│   ├── projects
│   ├── projects.json
│   └── users.json
└── notifications
    └── notifications.json
```
Which does not seem correct to me (see that `builds` folder, that is not part of the original repo, but generated by the CI inside the structure of the repo, which can lead to issues).

### PR DESCRIPTION
**Pull request by minuscorp**
Fixes #699

### CODE REVIEW

File: app/features/project/project_controller.rb [hunk 1]
```diff
@@ -22,9 +22,10 @@ class ProjectController < AuthenticatedControllerBase
 
       project = user_project_with_id(project_id: project_id)
       current_github_provider_credential = check_and_get_provider_credential
-
-      # Create random folder for checkout, prefixed with `manual_build`
-      checkout_folder = File.join(File.expand_path(project.local_repo_path), "manual_build_#{SecureRandom.uuid}")
+      # rubocop:disable Metrics/LineLength
```
Comment (KrauseFx) [inline app/features/project/project_controller.rb]: Can we split this in multiple lines? ;)

File: app/features/project/project_controller.rb [hunk 2]
```diff
@@ -22,9 +22,10 @@ class ProjectController < AuthenticatedControllerBase
...
+      # Create random folder for checkout, prefixed with `manual_build` or use the current_sha with the number of times we made a re-run for this commit.
+      checkout_folder = File.join(File.expand_path(project.local_repo_path), "manual_build_#{current_sha || SecureRandom.uuid}#{current_sha ? "_#{Dir[File.join(File.expand_path(project.local_repo_path), "*#{current_sha}*")].count}" : ''}")
```
Comment (taquitos) [inline app/features/project/project_controller.rb]: Wow that's a long confusing line 😄 
Let's break this down into a couple lines so it's easier to understand.

Comment (KrauseFx) [reply app/features/project/project_controller.rb]: ☝️

File: app/features/project/project_controller.rb (hunk 1 above)
Comment (taquitos) [reply app/features/project/project_controller.rb]: ☝️

**Pull request merged**